internal/dag/gatewayapi_processor.go

```diff
@@ -106,11 +118,82 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 		// Process all the routes that match this Gateway.
 		for _, matchingRoute := range matchingRoutes {
-			p.computeHTTPRoute(matchingRoute)
+			p.computeHTTPRoute(matchingRoute, listenerSecret)
 		}
 	}
 }
 
+func (p *GatewayAPIProcessor) validGatewayTLS(listener gatewayapi_v1alpha1.Listener) *Secret {
+
+	// Validate the CertificateRef is configured.
+	if listener.TLS.CertificateRef == nil {
+		p.Error(contour_api_v1.ConditionTypeTLSError, "SecretNotConfigured",
+			"Spec.VirtualHost.TLS Secret is not configured")
+		return nil
+	}
+
+	// Validate the correct protocol is specified.
+	if listener.Protocol != gatewayapi_v1alpha1.HTTPSProtocolType {
+		p.Error(contour_api_v1.ConditionTypeTLSError, "InvalidProtocol",
+			"Spec.VirtualHost.Protocol %q is not valid.")
+		return nil
+	}
+
+	// Validate a v1.Secret is referenced which can be kind: secret & group: core.
+	// ref: https://github.com/kubernetes-sigs/gateway-api/pull/562
+	if !isSecretRef(listener.TLS.CertificateRef) {
+		p.Error(contour_api_v1.ConditionTypeTLSError, "InvalidReference",
+			"Spec.VirtualHost.TLS Secret must be type v1.Secret")
+		return nil
+	}
+
+	listenerSecret, err := p.source.LookupSecret(types.NamespacedName{Name: listener.TLS.CertificateRef.Name, Namespace: p.source.gateway.Namespace}, validSecret)
+	if err != nil {
+		p.Errorf(contour_api_v1.ConditionTypeTLSError, "SecretNotValid",
+			"Spec.VirtualHost.TLS Secret %q is invalid: %s", listener.TLS.CertificateRef.Name, err)
+		return nil
+	}
+	return listenerSecret
+}
+
+func isSecretRef(certificateRef *gatewayapi_v1alpha1.LocalObjectReference) bool {
+	return strings.ToLower(certificateRef.Kind) == "secret" && strings.ToLower(certificateRef.Group) == "core"
+}
+
+func (p *GatewayAPIProcessor) computeHosts(route *gatewayapi_v1alpha1.HTTPRoute) ([]string, []error) {
```

Comment:
Do we have tests for all the codepaths in here? Esp. the wildcards etc.

Reply:
added tests